torch/utils/data/dataloader.py

```diff
@@ -242,11 +269,23 @@ def _shutdown_workers(self):
         if not self.shutdown:
             self.shutdown = True
             self.done_event.set()
+            # if worker_manager_thread is waiting to put
+            if not self.data_queue.empty():
+                self.data_queue.get()
             for _ in self.workers:
                 self.index_queue.put(None)
+            # if all workers hang, no None is showed to worker_manager_thread,
+            # we put None to let worker_manager_thread exit
+            if self.worker_result_queue.empty():
+                self.worker_result_queue.put(None)
+
+    def _remove_handers(self):
+        if self.handlers_set:
+            self.handlers_set = not _remove_main_signal_handlers_for_workers(self.worker_pids)
```

Comment:
Can you just make the method return nothing, and always set the value to `False`? It's not immediately clear what the return value is supposed to be (as far as I see it's a `True` constant)